parse/BuildingsParser.cpp

```diff
@@ -48,21 +40,23 @@ namespace {
             qi::_b_type _b;
             qi::_c_type _c;
             qi::_d_type _d;
+            qi::_pass_type _pass;
             qi::_r1_type _r1;
             qi::eps_type eps;
 
             const parse::lexer& tok = parse::lexer::instance();
 
             building_type
                 =   tok.BuildingType_
-                >   parse::detail::label(Name_token)                > tok.string        [ _a = _1 ]
+                >   parse::detail::label(Name_token)
+                >   tok.string        [ _pass = is_unique_(_r1, "BuildingType", _1), _a = _1 ]
```

Comment:
Fails with boost 1.64 (and likely 1.63) without `-DBOOST_PHOENIX_NO_VARIADIC_EXPRESSION`: 
```tty
[ 23%] Linking CXX shared library libfreeorioncommon.so
In file included from /opt/projects/boost/1_64_0/include/boost/phoenix/function/detail/cpp03/preprocessed/function_operator.hpp:12:0,
                 from /opt/projects/boost/1_64_0/include/boost/phoenix/function/detail/cpp03/function_operator.hpp:6,
                 from /opt/projects/boost/1_64_0/include/boost/phoenix/function/function.hpp:58,
                 from /opt/projects/boost/1_64_0/include/boost/phoenix/function.hpp:11,
                 from /opt/projects/boost/1_64_0/include/boost/spirit/include/phoenix_function.hpp:11,
                 from /opt/projects/boost/1_64_0/include/boost/spirit/home/support/terminal.hpp:18,
                 from /opt/projects/boost/1_64_0/include/boost/spirit/home/support/common_terminals.hpp:15,
                 from /opt/projects/boost/1_64_0/include/boost/spirit/home/lex/lexer/terminals.hpp:13,
                 from /opt/projects/boost/1_64_0/include/boost/spirit/home/lex/lexer.hpp:13,
                 from /opt/projects/boost/1_64_0/include/boost/spirit/home/lex.hpp:13,
                 from /opt/projects/boost/1_64_0/include/boost/spirit/home/lex/lexer_lexertl.hpp:15,
                 from /opt/projects/boost/1_64_0/include/boost/spirit/include/lex_lexertl.hpp:16,
                 from /opt/projects/freeorion/parse/Lexer.h:6,
                 from /opt/projects/freeorion/parse/ReportParseError.h:4,
                 from /opt/projects/freeorion/parse/ParseImpl.h:4,
                 from /opt/projects/freeorion/parse/BuildingsParser.cpp:3:
/opt/projects/boost/1_64_0/include/boost/phoenix/function/detail/cpp03/preprocessed/function_operator_10.hpp: In instantiation of ‘const typename boost::phoenix::detail::expression::function_eval<F, A0, A1, A2>::type boost::phoenix::function<F>::operator()(const A0&, const A1&, const A2&) const [with A0 = boost::phoenix::actor<boost::spirit::attribute<1> >; A1 = char [13]; A2 = boost::phoenix::actor<boost::spirit::argument<0> >; F = parse::detail::is_unique; typename boost::phoenix::detail::expression::function_eval<F, A0, A1, A2>::type = boost::phoenix::actor<boost::proto::exprns_::basic_expr<boost::phoenix::detail::tag::function_eval, boost::proto::argsns_::list4<boost::proto::exprns_::basic_expr<boost::proto::tagns_::tag::terminal, boost::proto::argsns_::term<parse::detail::is_unique>, 0l>, boost::phoenix::actor<boost::spirit::attribute<1> >, boost::proto::exprns_::basic_expr<boost::proto::tagns_::tag::terminal, boost::proto::argsns_::term<char*>, 0l>, boost::phoenix::actor<boost::spirit::argument<0> > >, 4l> >]’:
/opt/projects/freeorion/parse/BuildingsParser.cpp:52:83:   required from here
/opt/projects/boost/1_64_0/include/boost/phoenix/function/detail/cpp03/preprocessed/function_operator_10.hpp:57:76: error: invalid conversion from ‘const char*’ to ‘param_type {aka char*}’ [-fpermissive]
             return detail::expression::function_eval<F, A0 , A1 , A2>::make(f, a0 , a1 , a2);
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
In file included from /opt/projects/boost/1_64_0/include/boost/phoenix/core/argument.hpp:14:0,
                 from /opt/projects/boost/1_64_0/include/boost/phoenix/core.hpp:15,
                 from /opt/projects/boost/1_64_0/include/boost/spirit/include/phoenix_core.hpp:11,
                 from /opt/projects/boost/1_64_0/include/boost/spirit/home/support/terminal.hpp:17,
                 from /opt/projects/boost/1_64_0/include/boost/spirit/home/support/common_terminals.hpp:15,
                 from /opt/projects/boost/1_64_0/include/boost/spirit/home/lex/lexer/terminals.hpp:13,
                 from /opt/projects/boost/1_64_0/include/boost/spirit/home/lex/lexer.hpp:13,
                 from /opt/projects/boost/1_64_0/include/boost/spirit/home/lex.hpp:13,
                 from /opt/projects/boost/1_64_0/include/boost/spirit/home/lex/lexer_lexertl.hpp:15,
                 from /opt/projects/boost/1_64_0/include/boost/spirit/include/lex_lexertl.hpp:16,
                 from /opt/projects/freeorion/parse/Lexer.h:6,
                 from /opt/projects/freeorion/parse/ReportParseError.h:4,
                 from /opt/projects/freeorion/parse/ParseImpl.h:4,
                 from /opt/projects/freeorion/parse/BuildingsParser.cpp:3:
/opt/projects/boost/1_64_0/include/boost/phoenix/core/expression.hpp:63:21: note:   initializing argument 3 of ‘static boost::phoenix::expr_ext<Actor, Tag, A>::type boost::phoenix::expr_ext<Actor, Tag, A>::make(typename boost::call_traits<A>::param_type ...) [with Actor = boost::phoenix::actor; Tag = boost::phoenix::detail::tag::function_eval; A = {parse::detail::is_unique, boost::phoenix::actor<boost::spirit::attribute<1> >, char*, boost::phoenix::actor<boost::spirit::argument<0> >}; boost::phoenix::expr_ext<Actor, Tag, A>::type = boost::phoenix::actor<boost::proto::exprns_::basic_expr<boost::phoenix::detail::tag::function_eval, boost::proto::argsns_::list4<boost::proto::exprns_::basic_expr<boost::proto::tagns_::tag::terminal, boost::proto::argsns_::term<parse::detail::is_unique>, 0l>, boost::phoenix::actor<boost::spirit::attribute<1> >, boost::proto::exprns_::basic_expr<boost::proto::tagns_::tag::terminal, boost::proto::argsns_::term<char*>, 0l>, boost::phoenix::actor<boost::spirit::argument<0> > >, 4l> >]’
         static type make(typename call_traits<A>::param_type... a)
                     ^~~~
CMakeFiles/freeorionparseobj.dir/build.make:86: recipe for target 'CMakeFiles/freeorionparseobj.dir/parse/BuildingsParser.cpp.o' failed
```

Related previous workaround in #1271 (not that this is the only solution)